implementations/rust/vault/src/ffi/mod.rs

```diff
@@ -463,9 +464,8 @@ pub extern "C" fn ockam_vault_aead_aes_gcm_decrypt(
     plaintext_size: u32,
     plaintext_length: &mut u32,
 ) -> VaultError {
-    // FIXME: Additional data is not mandatory
-    check_buffer!(additional_data, additional_data_length);
-    check_buffer!(ciphertext_and_tag, plaintext_size);
+    check_buffer!(ciphertext_and_tag);
```

Comment:
ciphertext_and_tag should never be 0 length.  This check needs to remain.

```suggestion
    check_buffer!(ciphertext_and_tag, ciphertext_and_tag_length);
```